### PR TITLE
refactor, test(pool): Change `updatePosition` param

### PR DESCRIPTION
**Pull request by notJoon**
# Description

- Change `updatePosition` function's parameter to only take `ModifyPositionParams` type
- Add unit test for `modifyPosition`. this test covers both `update` and `modify` function

### CODE REVIEW

**Review by onlyhyde — APPROVED** [review]

**Review by onlyhyde — DISMISSED** [review]

**Review by r3v4s — APPROVED** [review]

File: pool/position_modify.gno [hunk 1]
```diff
@@ -11,13 +11,7 @@ import (
...
-	position := pool.updatePosition(
-		params.owner,
-		params.tickLower,
-		params.tickUpper,
-		params.liquidityDelta,
-		pool.slot0.tick,
-	)
+	position := pool.updatePosition(params)
 
 	var amount0, amount1 *i256.Int
```
Comment (dongwon8247) [inline pool/position_modify.gno:17]: Can you improve the nested if-else lines? Let's address them and avoid using it for better readability.

File: pool/position_modify.gno [hunk 2]
```diff
@@ -11,13 +11,7 @@ import (
 // modifyPosition updates a position in the pool and calculates the amount of tokens to be added or removed.
 // Returns positionInfo, amount0, amount1
 func (pool *Pool) modifyPosition(params ModifyPositionParams) (PositionInfo, *i256.Int, *i256.Int) {
```
Comment (dongwon8247) [inline pool/position_modify.gno:13]: This func should be simply about updating position. You can divide calculateTokenDeltas into another function.

**Review by dongwon8247 — CHANGES_REQUESTED** [review]

File: pool/position_modify.gno (hunk 1 above)
Comment (dongwon8247) [reply pool/position_modify.gno:17]: Checked you already did this in #402. Great job.

File: pool/position_modify.gno (hunk 2 above)
Comment (dongwon8247) [reply pool/position_modify.gno:13]: Checked you already did this in #402. Great job.

**Review by dongwon8247 — APPROVED** [review]

**Pull request merged**